packages/commonwealth/client/scripts/views/pages/notification_settings/index.tsx

```diff
@@ -47,6 +58,47 @@ const NotificationSettingsPage = () => {
     app.user.notifications.isLoaded.once('redraw', forceRerender);
   }, [app?.user.notifications, app.user.emailInterval]);
 
+  useEffect(() => {
+    // bundled snapshot subscriptions
+    const bundledSnapshotSubs = extractSnapshotProposals(
+      app.user.notifications.discussionSubscriptions
+    );
+    const snapshotIds = Object.keys(bundledSnapshotSubs);
+
+    const getTheSpace = async () => {
+      try {
+        // Fetch spaces for all snapshotIds concurrently
+        const spacePromises = snapshotIds.map(async (snapshotId) => {
```

Comment:
Given that these are rate-limited public APIs we should do our best to optimize these requests. Can we not use the [`getMultipleSpaces` endpoint][1]? Additionally, since it is a GraphQL API, we should limit the data we fetch to just the avatar and the snapshot id.

[1]: https://docs.snapshot.org/tools/api#spaces

Reply:
I'll do some exploration on this and report back here. We may be able to, i'll just have to look at the schema and make sure it's a 1:1

Reply:
Updated to query the correct endpoint.